lib/rubocop/cop/style/yaml_file_read.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Checks for the use of `YAML.load`, `YAML.safe_load`, and `YAML.parse` with
+      # `File.read` argument.
+      #
+      # NOTE: `YAML.safe_load_file` was introduced in Ruby 3.0.
+      #
+      # @example
+      #
+      #   # bad
+      #   YAML.load(File.read(path))
```

Comment:
Btw, I'm guessing we should be explicit and mention that it's fine to pass a file data explicitly if you already happen to have it. (e.g. a good example)

I'm also wondering if this not a target for a broader cop - I assume there are more methods that work with a file object, file data or a file path in ruby.